python/ray/experimental/state/state_cli.py

```diff
@@ -143,25 +145,138 @@ def get_api_server_url() -> str:
     return api_server_url
 
 
+def get_table_output(state_data: List):
+    time = datetime.now()
+    header = "=" * 8 + f" List: {time} " + "=" * 8
+    headers = []
+    table = []
+    for data in state_data:
+        for key, val in data.items():
+            if isinstance(val, dict):
+                data[key] = yaml.dump(val, indent=2)
+        headers = sorted([key.upper() for key in data.keys()])
+        table.append([data[header.lower()] for header in headers])
+    return f"""
+{header}
+Stats:
+------------------------------
+Total: {len(state_data)}
+
+Table:
+------------------------------
+{tabulate(table, headers=headers, showindex=True, tablefmt="plain", floatfmt=".3f")}
+"""
+
+
 def output_with_format(
-    state_data: Union[dict, list], format: AvailableFormat = AvailableFormat.DEFAULT
+    state_data: List, format: AvailableFormat = AvailableFormat.DEFAULT
 ):
     # Default is yaml.
     if format == AvailableFormat.DEFAULT:
-        return yaml.dump(state_data, indent=4, explicit_start=True)
+        return get_table_output(state_data)
     if format == AvailableFormat.YAML:
         return yaml.dump(state_data, indent=4, explicit_start=True)
     elif format == AvailableFormat.JSON:
         return json.dumps(state_data)
     elif format == AvailableFormat.TABLE:
-        raise NotImplementedError("Table formatter is not implemented yet.")
+        return get_table_output(state_data)
     else:
         raise ValueError(
             f"Unexpected format: {format}. "
             f"Supported formatting: {_get_available_formats()}"
         )
 
 
+def format_summary_output(state_data: Dict, *, resource: StateResource):
```

Comment:
what's this `*` for?

Reply:
this forces the kwarg (e.g., `resource ` should be kwargs although it doesn't have the default value)